compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -467,6 +467,24 @@ private void checkXMLNamespacePrefixes(List<BLangXMLElementFilter> filters) {
         }
     }
 
+    private int targetMemberTypeTag(BFiniteType finiteType) {
+        for (int i = TypeTags.SIGNED32_INT; i <= TypeTags.UNSIGNED8_INT; i++) {
+            for (BLangExpression valueExpr : finiteType.getValueSpace()) {
+                if (valueExpr.getBType().tag == i) {
+                    return i;
+                }
+            }
+        }
+        for (int i = TypeTags.INT; i <= TypeTags.DECIMAL; i++) {
+            for (BLangExpression valueExpr : finiteType.getValueSpace()) {
+                if (valueExpr.getBType().tag == i) {
+                    return i;
+                }
+            }
+        }
+        return 0;
```

Comment:
Below failing at runtime
```ballerina
type Mat 1d|"dummy";

function dummyTest() {
     Mat f1 = 1;
     boolean res = f1 is 1d;
}
```

Reply:
Shall I create a separate issue for this?. Because this issue does not relate to the type. right?
```
[2021-11-05 08:37:12,785] SEVERE {b7a.log.crash} - Character d is neither a decimal digit number, decimal point, nor "e" notation exponential mark. 
java.lang.NumberFormatException: Character d is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
        at java.base/java.math.BigDecimal.<init>(BigDecimal.java:518)
        at java.base/java.math.BigDecimal.<init>(BigDecimal.java:852)
        at io.ballerina.runtime.internal.values.DecimalValue.<init>(DecimalValue.java:71)
        at test3.dummyTest(test3.bal:5)
        at test3.main(test3.bal:10)
        at $_init.$lambda$main$(.)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:610)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:323)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:291)
        at java.base/java.lang.Thread.run(Thread.java:829)
 

```

Reply:
@chiranSachintha https://github.com/ballerina-platform/ballerina-lang/issues/33266 Can you check with Hinduja regarding this? Seems related issue d been fixed earlier